src/UIComponents.cpp

```diff
@@ -0,0 +1,91 @@
+#include "UIComponents.hpp"
+
+UIEntity::~UIEntity()
+{
+  entity->detach((IObserver*)this);
+};
+
+UIEntity::UIEntity(std::shared_ptr<HAEntity> _entity, lv_obj_t* _parent)
+{
+  parentContainer = _parent;
+  entity = _entity;
+  entity->attach((IObserver*)this);
+};
+
+UIButton::UIButton(std::shared_ptr<HAEntity> _entity, lv_obj_t* _parent) :
+  UIEntity(_entity, _parent)
+{
+  // START BUTTON EXAMPLE
+  btn = lv_btn_create(_parent); /*Add a button the current screen*/
+  lv_obj_set_size(btn, 240, 50);
+  lv_obj_center(btn);
+  lv_obj_add_flag(btn, LV_OBJ_FLAG_CHECKABLE);
+  lv_obj_add_event_cb(btn, UIButton::btn_press_cb, LV_EVENT_ALL, reinterpret_cast<void*>(&entity)); /*Assign a callback to the button*/
```

Comment:
all the magic in this line is duplicated in UISwitch, maybe we should make some macro/template that makes this less typing

Reply:
I agree, but i think UIButton and UISwitch are very much the same. I'm not sure if other UIentities are different or not, so for me it feels to early to change this now. Added a note though!